sw/device/lib/dif/dif_hmac.c

```diff
@@ -0,0 +1,400 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+
+#include "hmac_regs.h"  // Generated.
+
+/**
+ * Load `key` into `hmac`.
+ */
+static void load_key(const dif_hmac_t *hmac, const uint8_t *key) {
+  mmio_region_memcpy_to_mmio32(hmac->base_addr, HMAC_KEY0_REG_OFFSET, key,
+                               HMAC_PARAM_NUMWORDS * sizeof(uint32_t));
```

Comment:
Since this requires the addresses of `HMAC_KEY0_REG_OFFSET` ... `HMAC_KEY7_REG_OFFSET` to be consecutive, you may want to add an assert here or in the unit test.

Reply:
`_Static_assert` is probably the way forwards, I'm happy for that to happen in a follow-up.